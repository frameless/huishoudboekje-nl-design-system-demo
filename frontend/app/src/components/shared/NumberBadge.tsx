import {Badge, BadgeProps} from "@chakra-ui/react";
import React from "react";

type NumberBadgeProps = BadgeProps & {
	count: number
};

const NumberBadge: React.FC<NumberBadgeProps> = ({count, ...props}) => {
	return (
		<Badge fontSize={"sm"} p={1} colorScheme={"secondary"} {...props}>
			{count > 99 ? "99+" : count}
		</Badge>
	);
};

export default NumberBadge;