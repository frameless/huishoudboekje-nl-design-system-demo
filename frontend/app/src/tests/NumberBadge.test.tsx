import {act} from "@testing-library/react";
import React from "react";
import {render, unmountComponentAtNode} from "react-dom";
import NumberBadge from "../components/shared/NumberBadge";
import pretty from "pretty";

let container: HTMLDivElement | null = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container!);
    container!.remove();
    container = null;
});

describe("Numberbadge", () => {

    it("shows NumberBage count", () => {

        act(() => {
            render(<NumberBadge count={9} />, container);
        });

        expect(pretty(container?.innerHTML)).toMatchSnapshot();
    });

    it("shows NumberBage count more than 99", () => {

        act(() => {
            render(<NumberBadge count={9999} />, container);
        });

        expect(pretty(container?.innerHTML)).toMatchSnapshot();
    });
})

