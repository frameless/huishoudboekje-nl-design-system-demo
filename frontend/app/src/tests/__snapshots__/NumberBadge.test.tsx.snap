// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numberbadge shows NumberBage count 1`] = `"<span class=\\"chakra-badge css-ni0re5\\">9</span>"`;

exports[`Numberbadge shows NumberBage count more than 99 1`] = `"<span class=\\"chakra-badge css-ni0re5\\">99+</span>"`;
